### DISCUSSION TITLE
prisma db introspection comments maps for column names which are in number format

### DISCUSSION THREAD

**Discussion by njoshi10** [prisma/prisma#20376] [Q&A]
I'm using mysql db, and have a table whose column names are 1, 2, 3... Prisma doesn't support such column names, so I changed my model to this.
```prisma
model dc_map {
  id           Int       @id @default(autoincrement()) @db.UnsignedInt
  region       String
  one          String    @map("1")
  two          String?   @map("2")
  three        String?   @map("3")
  four         String?   @map("4")
  five         String?   @map("5")
  six          String?   @map("6")
  seven        String?   @map("7")
  eight        String?   @map("8")
  nine         String?   @map("9")
  ten          String?   @map("10")
...
}
```
When I generate prisma client with these mappings, my queries run fine but as soon as I run 'prisma db pull', it comments on these mappings saying:

```
/// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // one String @map("1")
  /// This field was commented out because of an invalid name. Please provide a valid one that matches [a-zA-Z][a-zA-Z0-9_]*
  // two String? @map("2").. 
```
and so on

whereas my terminal says two different things,

```
These fields were enriched with @map information taken from the previous Prisma schema.
- Model "dc_map", field: "one"
- Model "dc_map", field: "two"
- Model "dc_map", field: "three"
- Model "dc_map", field: "four"
- Model "dc_map", field: "five"
- Model "dc_map", field: "six"
- Model "dc_map", field: "seven"
- Model "dc_map", field: "eight"
- Model "dc_map", field: "nine"
- Model "dc_map", field: "ten"

These fields were commented out because their names are currently not supported by Prisma. Please provide valid ones that match [a-zA-Z][a-zA-Z0-9_]* using the @map attribute.
- Model: "dc_map"
  Field(s): "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"
```
I am currently using prisma and prisma client version 4.8.0

I tried:

deleting node_modules and re-running npm i.
Restart VS Code, tried with another project with the same database, and faced the same issue.
tried downgrading my version from 4.8.0 to 4.6.1.
What I am expecting: Prisma ideally should support anything as a mapped coulmn name which matches [a-zA-Z][a-zA-Z0-9_]* maps like `one     String    @map("1") `work fine after the client gets generated. But I expect that it should not comment on these mappings with warnings saying that those names are invalid.

Other observations:
I have a pretty old project with the same schema file and prisma version 4.6.1 where when I run prisma db pull, my dc_map model doesn't get disturbed at all.

**Comment by janpio** [comment]
That sounds like a bug. Can you please open a bug report, optimally include the SQL and schema that you are experiencing this with? Thanks.

**Reply by ludralph** [reply]
@njoshi10 I can confirm that this happens as well with Prisma version 5. Can you please use this [link](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) to open a bug report?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
